### PR TITLE
Change stable channel downloads to use only github api and links

### PR DESCRIPTION
**Pull request by Daenyth**
## Related Ticket(s)
- Fixing #2502

## What will change with this Pull Request?
- We don't hit bintray at all for stable channel updates
- Uses fewer network calls by using the assets list from github api

## Screenshot
screen shot 2017-03-19 at 9 11 35 am

### CODE REVIEW

File: cockatrice/src/releasechannel.cpp [hunk 1]
```diff
@@ -68,18 +65,13 @@ bool ReleaseChannel::downloadMatchesCurrentOS(QVariantMap build)
...
 
-    auto fileName = build["name"].toString();
-    // Checking for .zip is a workaround for the May 6th 2016 release
-    auto zipName = arch + ".zip";
     auto exeName = arch + ".exe";
-    auto zipDebugName = devSnapshotEnd + ".zip";
     auto exeDebugName = devSnapshotEnd + ".exe";
-    return (fileName.endsWith(exeName) || fileName.endsWith(zipName) ||
-        fileName.endsWith(exeDebugName) || fileName.endsWith(zipDebugName));
+    return (fileName.endsWith(exeName) || fileName.endsWith(exeDebugName));
```
Comment (tooomm) [inline cockatrice/src/releasechannel.cpp]: why 2x Debug?

**Comment by Daenyth** [comment]
Currently broken - hitting 'yes' crashes and gives me this file opened
```
{"url":"https://api.github.com/repos/Cockatrice/Cockatrice/releases/assets/3389752","id":3389752,"name":"Cockatrice_Thopter_Pie_Network_osx.dmg","label":null,"uploader":{"login":"ZeldaZach","id":7460172,"avatar_url":"https://avatars3.githubusercontent.com/u/7460172?v=3","gravatar_id":"","url":"https://api.github.com/users/ZeldaZach","html_url":"https://github.com/ZeldaZach","followers_url":"https://api.github.com/users/ZeldaZach/followers","following_url":"https://api.github.com/users/ZeldaZach/following{/other_user}","gists_url":"https://api.github.com/users/ZeldaZach/gists{/gist_id}","starred_url":"https://api.github.com/users/ZeldaZach/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/ZeldaZach/subscriptions","organizations_url":"https://api.github.com/users/ZeldaZach/orgs","repos_url":"https://api.github.com/users/ZeldaZach/repos","events_url":"https://api.github.com/users/ZeldaZach/events{/privacy}","received_events_url":"https://api.github.com/users/ZeldaZach/received_events","type":"User","site_admin":false},"content_type":"application/x-diskcopy","state":"uploaded","size":35024018,"download_count":87,"created_at":"2017-03-14T04:00:28Z","updated_at":"2017-03-14T04:00:39Z","browser_download_url":"https://github.com/Cockatrice/Cockatrice/releases/download/2017-03-14-Release/Cockatrice_Thopter_Pie_Network_osx.dmg"}
```

**Comment by Daenyth** [comment]
Ah, it's not a crash, it's expected behavior (stupid)

`//Try to open the installer. If it opens, quit Cockatrice`

**Comment by tooomm** [comment]
Can you include a short description of how the update process works now?
It's a very critical function of the app^^

**Comment by Daenyth** [comment]
Same as before if you're on stable channel

**Comment by Daenyth** [comment]
Fixed downloading json instead of the file.
screen shot 2017-03-19 at 9 27 13 am

**Comment by tooomm** [comment]
So the internal update check works also the same?
(that was the question... not the steps for the user)

**Comment by Daenyth** [comment]
- Get latest release from github api
- Look for an os-appropriate file from that release's assets
- Get tag name -> commit id from github api. Compare those same as before to see if update is needed

Basically the same process except the old version had an extra step where it would take the commit and look for a file on bintray - now finding the file is part of the first request handler

File: cockatrice/src/releasechannel.cpp (hunk 1 above)
Comment (ZeldaZach) [inline cockatrice/src/releasechannel.cpp]: it doesn't really matter what the variable name is called, it's pretty clear what it's use is

File: cockatrice/src/dlg_update.cpp [hunk 2]
```diff
@@ -163,6 +164,7 @@ void DlgUpdate::downloadSuccessful(QUrl filepath) {
     if (QDesktopServices::openUrl(filepath))
     {
         QMetaObject::invokeMethod((MainWindow*) parent(), "close", Qt::QueuedConnection);
+        qInfo() << "Opened downloaded update file successfully - closing Cockatrice";
```
Comment (ZeldaZach) [inline cockatrice/src/dlg_update.cpp]: So this showing out of scope on the linux builds is a bug in QT < 5.4. Do we compile against that or lower for the linux builds?

**Comment by Daenyth** [comment]
> If i'm on a dev build and switch to stable, it will tell me to "update" to Pie Network, even though, for me, that is a downgrade

I think that's fine for now. When we do semver it will work better since the snapshot can be something like x.y.z-git01234abcd, or whatever we want to make it > x.y.z

>Linux builds are failing to due a bug, so this is fine to go whenever you're ready on it.

Annoying; I think the next-next release should get travis building on modern ubuntu with modern qt via docker. PITA but we have to do it eventually.

> Do you think we should upload this version to bintray as a new "release" and then plan around github releases? I think I'd like to sort out the dev snapshots before we do that, but that's just me.

Yes I want to do that but as you say, not until we figure out dev snapshots

**Comment by Daenyth** [comment]
`/home/travis/build/Cockatrice/Cockatrice/cockatrice/src/releasechannel.cpp:153:11: error: ‘qInfo’ was not declared in this scope`

I think I can fix this with an extra include

**Comment by Daenyth** [comment]
Zach what's the bug? If those travis builds won't work we lose trusty deb builds

File: cockatrice/src/releasechannel.cpp (hunk 1 above)
Comment (tooomm) [inline cockatrice/src/releasechannel.cpp]: proper name > bad name only because its use is clear

**Comment by tooomm** [comment]
Another thing I run into while bintray wasn't working...

I always received two stacked error dialogs - is this still the case?
Maybe somebody can adjust their local builds and force an error.

update error since bintray prob

**Comment by ZeldaZach** [comment]
Status of this?

**Comment by Daenyth** [comment]
I haven't had time to address feedback yet. If you want to throw some commits on the branch go for it, otherwise it'll have to wait until I have time

**Comment by woogerboy21** [comment]
> I always received two stacked error dialogs - is this still the case?

@tooomm 
Are you building on your own or pulling from appveyor?

**Comment by tooomm** [comment]
>Are you building on your own or pulling from appveyor?

Appveyor all the time!

**Comment by ZeldaZach** [comment]
Since i fixed the popup issue, what's left on this @tooomm ?

**Comment by tooomm** [comment]
https://github.com/Cockatrice/Cockatrice/pull/2515#pullrequestreview-27797792

`Open download page` (if on release channe)l should have the direct link to latest release.

**Comment by ZeldaZach** [comment]
Fixed @tooomm . I'm ready to merge this.

**Pull request merged**